src/aarch64/ffi.c

```diff
@@ -743,6 +877,13 @@ ffi_call_int (ffi_cif *cif, void (*fn)(void), void *orig_rvalue,
    * points just after the alloca with the curretn stack bounds.
    */
   stack = __builtin_cheri_address_set(__builtin_cheri_stack_get(), (ptraddr_t)stack);
+  /*
+   * Align the stack so that the c9 capability to on-stack function arguments
+   * can be bounded exactly to the number of bytes occupied by them.
+   */
+  stack = __align_up(stack,
+		     ~__builtin_cheri_representable_alignment_mask(stack_argsize) +
+		     1);
```

Comment:
Defer this step till after the memory arguments have been allocated.

Reply:
We need `stack` before we store arguments.